### PR TITLE
Update Parameters and Value type consistency between packages inside Navigation 2

### PR DESCRIPTION
**Pull request by drhieu**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (#1945 on Foxy Devel) |
| Primary OS tested on | (Ubuntu 20.04) |
| Robotic platform tested on | (Others) |

---

## Description of contribution in a few bullet points

*Update parameters name in amcl to match other packages in navigation. As well as, how you can initialize its value in config files.
*Now you can use both "quote" and unquote values for frames in config file like other packages.

## Description of documentation updates required from your changes

* This is ABI braking because of the parameters name changes. 
* Update to navigation.ros.org is needed. 
---

### CODE REVIEW

**Comment by SteveMacenski** [comment]
There's an (odd) linter error

```
    build/include_what_you_use [4] (/opt/overlay_ws/src/navigation2/nav2_waypoint_follower/src/waypoint_follower.cpp:48) - nav2_waypoint_follower.cpplint

    Add #include <vector> for vector<>
```

Please add this so that CI passes and we can merge

File: nav2_amcl/include/nav2_amcl/amcl_node.hpp [hunk 1]
```diff
@@ -217,12 +217,12 @@ class AmclNode : public nav2_util::LifecycleNode
   double alpha3_;
```
Comment (SteveMacenski) [inline nav2_amcl/include/nav2_amcl/amcl_node.hpp:217]: You must also update the docs/parameters/params.md file with these updates

**Comment by SteveMacenski** [comment]
Or pull in main, so that you can get the update that Ruffin already fixed this with. Looks like your branch is still out of date

**Comment by mikeferguson** [comment]
@SteveMacenski should we update the title of this PR - seems strange to have "foxy devel" in the title when it's going into main (and probably never getting ported to foxy)

**Comment by SteveMacenski** [comment]
@drhieu any update?

**Comment by drhieu** [comment]
I'll try to make these changes within this week. I'm a little bit busy with
work at the moment.

**Comment by SteveMacenski** [comment]
Closing from inactivity.

**Pull request closed without merging**